jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -284,6 +308,38 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
 
         return collapsed_spec
 
+    def cone_aperture(self):
+        if not self._app._jdaviz_helper._loaded_mask_cube:
+            snackbar_message = SnackbarMessage(
+                "Cannot create cone aperture without valid mask cube loaded.",
+                color="error",
+                sender=self)
+            self.hub.broadcast(snackbar_message)
+            return
+        # Retrieve mask cube and create array to represent the cone mask
+        mask_cube = self._app._jdaviz_helper._loaded_mask_cube.get_object(cls=Spectrum1D,
```

Comment:
I still don't understand why the mask handling here is different from the "not cone" case. Why the inconsistency?

Reply:
We use a different mask handling because the aperture of the cone is different at each slice, where as when using a spatial subset that mask is the same throughout the cube.

Reply:
But mask cube was not even consider for the other case.

Reply:
Because the mask is provided by the spatial subset. The cone aperture just uses the center and radius from the subset and calculates the mask based on the mask cube.